### PR TITLE
Webcam Settings modal for choosing a webcam

### PR DESCRIPTION
**Pull request by MaximKhlobystov**
The modal appears after clicking `Share Webcam`.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/video-preview/component.jsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+      video: {
+        deviceId: webcamValue ? { exact: webcamValue } : undefined,
+      },
+    };
+    this.stopTracks();
+    navigator.mediaDevices.getUserMedia(constraints).then((stream) => {
+      this.video.srcObject = stream;
+      this.deviceStream = stream;
+    }).catch((error) => {
+      console.log(error);
```
Comment (BobakOftadeh) [inline bigbluebutton-html5/imports/ui/components/video-preview/component.jsx]: The client side logger could be used to log this error message. [http://docs.bigbluebutton.org/html/html5-dev.html#client-logger](url)

**Comment by Tainan404** [comment]
The modal stay resizing his height, because the video height is changing, could set one default height and only resize the video?

peek 2018-11-14 15-34

**Comment by Tainan404** [comment]
Is missing the rounded border compared with the proposed design.

Your PR:
screenshot from 2018-11-14 15-49-37

The design
screenshot from 2018-11-14 15-49-24

The video on video dock
screenshot from 2018-11-14 15-49-48

**Comment by Tainan404** [comment]
weird state of webcams:

peek 2018-11-16 16-00

**Comment by KDSBrowne** [comment]
I think you forgot to add `document.removeEventListener('beforeunload', this.unshareWebcam);` in `componentWillUnmount` of the `video-provider`.

**Pull request merged**